### ISSUE TITLE
Update Omada Controller

### ISSUE THREAD

**Issue by goozoon** [monsn0/omada-installer#10]
Great script, thank you very much.

When the new version of Omada Controller is released, just re-run the script?

Thank you

**Comment by monsn0** [comment]
You're welcome! :)

The script is currently designed to be ran as a new install only.
I should probably add a check if the Omada controller is already installed...

You'll update the controller from within the web portal under `Settings -> System Settings`
It is recommended to take a backup or snapshot prior to performing any upgrades.

---
![image](https://github.com/monsn0/omada-installer/assets/27322762/6edfadd1-905f-49f7-ae62-7c6f362d8c4a)

**Comment by goozoon** [comment]
Would be nice, fo sure.
But im thinking.. i installed it the same day i wrote here, so 2 days ago. I got 5.13.22 version installed and right after got a notification about new version. So i thought they released new version just in time of my installation.. But not... 5.13.23 Published Date: 2024-01-11

So there might be something wrong with a script. Can you check, please?

**Comment by monsn0** [comment]
Thanks for bringing up the version issue.
@zurno found this was due to TP-Link changing the casing of the latest version filename and was fixed on commit a419a87
I just cherry-picked it from PR #9 as another commit is included that I do not want to merge (yet).

**Comment by goozoon** [comment]
So i just run this portion of the new script (with 'i')

```bash
echo "[+] Downloading the latest Omada Software Controller package"
OmadaPackageUrl=$(curl -fsSL https://www.tp-link.com/us/support/download/omada-software-controller/ | grep -oPi '<a[^>]*href="\K[^"]*Linux_x64.deb[^"]*' | head -n 1)
wget -qP /tmp/ $OmadaPackageUrl
echo "[+] Installing Omada Software Controller"
dpkg -i /tmp/$(basename $OmadaPackageUrl) &> /dev/null

hostIP=$(hostname -I | cut -f1 -d' ')
echo -e "\e[0;32m[~] Omada Software Controller has been successfully installed! :)\e[0m"
echo -e "\e[0;32m[~] Please visit https://${hostIP}:8043 to complete the inital setup wizard.\e[0m\n"
```

and im still: Version:5.13.22

**Comment by zurno** [comment]
What OS version are you running? I have always been able to just install over an old install without any problem.

**Comment by goozoon** [comment]
> What OS version are you running? I have always been able to just install over an old install without any problem.

[+] Verifying supported OS
[~] Operating System: Ubuntu 22.04.3 LTS

**Comment by zurno** [comment]
I think the script leaves the downloaded file in the /tmp/ folder, maybe try clearing that out and run it again?

**Comment by goozoon** [comment]
> I think the script leaves the downloaded file in the /tmp/ folder, maybe try clearing that out and run it again?

5.13.23 is in there, strange
no previous version, so some other cache?

**Comment by zurno** [comment]
I am honestly not completely sure if that folder automatically removes things at some point. I just noticed it having multiple files when I was trying to get a Debian install to work. That was also when I noticed the version case issue.

**Comment by zurno** [comment]
Is your install directly on a machine, in a VoM, or a container?

**Comment by zurno** [comment]
If you are feeling adventurous, you could try the version of the script I was testing with, I unsilenced a lot of things for the Debian testing, but Ubuntu should function the same as before.

curl -sS https://raw.githubusercontent.com/zurno/omada-installer/testing/install-omada-controller.sh | sudo bash

**Comment by monsn0** [comment]
dpkg needs to run as root, the script assumes you are running the entire script as root.

Either place the entire snippet into a script file and execute the script as root or at least in your snippet above, execute the line with dpkg with sudo:

```bash
echo "[+] Downloading the latest Omada Software Controller package"
OmadaPackageUrl=$(curl -fsSL https://www.tp-link.com/us/support/download/omada-software-controller/ | grep -oPi '<a[^>]*href="\K[^"]*Linux_x64.deb[^"]*' | head -n 1)
wget -qP /tmp/ $OmadaPackageUrl
echo "[+] Installing Omada Software Controller"
sudo dpkg -i /tmp/$(basename $OmadaPackageUrl) &> /dev/null

hostIP=$(hostname -I | cut -f1 -d' ')
echo -e "\e[0;32m[~] Omada Software Controller has been successfully installed! :)\e[0m"
echo -e "\e[0;32m[~] Please visit https://${hostIP}:8043 to complete the inital setup wizard.\e[0m\n"
```

**Comment by goozoon** [comment]
> dpkg needs to run as root, the script assumes you are running the entire script as root.
> 
> Either place the entire snippet into a script file and execute the script as root or at least in your snippet above, execute the line with dpkg with sudo:
> 
> ```shell
> echo "[+] Downloading the latest Omada Software Controller package"
> OmadaPackageUrl=$(curl -fsSL https://www.tp-link.com/us/support/download/omada-software-controller/ | grep -oPi '<a[^>]*href="\K[^"]*Linux_x64.deb[^"]*' | head -n 1)
> wget -qP /tmp/ $OmadaPackageUrl
> echo "[+] Installing Omada Software Controller"
> sudo dpkg -i /tmp/$(basename $OmadaPackageUrl) &> /dev/null
> 
> hostIP=$(hostname -I | cut -f1 -d' ')
> echo -e "\e[0;32m[~] Omada Software Controller has been successfully installed! :)\e[0m"
> echo -e "\e[0;32m[~] Please visit https://${hostIP}:8043 to complete the inital setup wizard.\e[0m\n"
> ```

You are right, it have to be run with SUDO, it thrown an error in the background (original script)
So made a new script with just this snipped and executed - got the latest version

Today it wa not running, service was stopped, so made this:

```bash
me@nuc:~$ sudo systemctl enable tpeap
tpeap.service is not a native service, redirecting to systemd-sysv-install.
Executing: /lib/systemd/systemd-sysv-install enable tpeap
```

**Comment by monsn0** [comment]
Fixed with PR #9
Added check for root on commit 74db4e4
